src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java

```diff
@@ -299,42 +383,99 @@ public void simulationPeriodic() {
     pivotArmMech.setAngle(Radians.of(pivotSim.getAngleRads()).in(Degrees));
 
     switch (state) {
-      case Idle:
-        if (MathUtil.isNear(
-                pivotSim.getAngleRads(), kPivotExtendAngle.in(Radians), kPivotTolerance.in(Radians))
-            && Math.abs(intakeMotor.get()) > 0
-            && Math.abs(conveyorMotor.get()) > 0) {
-          timer.start();
+      case IDLE:
+        sensors.setSimState(CoralEnum.NO_CORAL);
+
+        if (atSetpoint(kPivotExtendAngle) && intakeMotor.get() > 0) {
+          t.start();
         } else {
-          timer.stop();
+          t.stop();
         }
 
-        if (timer.hasElapsed(2)) {
-          simbeam.set(true);
-          state = IntakeState.atSensor;
-          timer.reset();
+        if (t.hasElapsed(0.5)) {
+          state = IntakeState.FLOOR_INTAKE;
         }
-        break;
-      case atSensor:
-        if (Math.abs(conveyorMotor.get()) > 0) {
-          if (!timer.isRunning()) {
-            timer.start();
-          }
+
+        if (atSetpoint(kcoralStation) && intakeMotor.get() > 0) {
+          t.start();
         } else {
-          if (timer.isRunning()) {
-            timer.stop();
-          }
+          t.stop();
         }
 
-        if (timer.hasElapsed(2)) {
-          simbeam.set(false);
-          state = IntakeState.coralLoaded;
-          timer.reset();
+        if (t.hasElapsed(0.5)) {
+          state = IntakeState.HP_CORAL_INTAKE;
         }
+
+        if (atSetpoint(kalgae) && intakeMotor.get() < 0) {
+          t.start();
+        } else {
+          t.stop();
+        }
+
+        if (t.hasElapsed(0.5)) {
+          state = IntakeState.ALGAE_INTAKE;
+        }
+      case FLOOR_INTAKE:
+        sensors.setSimState(CoralEnum.CORAL_NOT_ALIGNED);
```

Comment:
Instead of CORAL_NOT_ALIGNED, you could do a random pick of CORAL_TOO_CLOSE or CORAL_TOO_FAR
`Math.random() > 0.5 ? CORAL_TOO_CLOSE : CORAL_TOO_FAR;`